make/PreInitSupport.gmk

```diff
@@ -251,12 +251,16 @@ endef
 # Param 2: The SPEC file to use.
 define DefineMainTargets
 
+  # We need to include the given SPEC file to setup TOPDIR properly
+  SPEC_FILE := $(strip $2)
+  -include $$(SPEC_FILE)
```

Comment:
Is it safe to include SPEC here? I thought the point of PreInit.gmk was that it didn't load it.

Reply:
Yeah, it is a bit scary. I don't really like it either. I tested this and it seemed okay, but we are not sure we are including the proper spec at this point, since we can pick an arbitrary spec file for e.g. tab expansion. I can't really say what could go wrong, but it is more of the lines "this is not obviously wrong" rather than "this is obviously right".

I also toyed with the idea of extracting the TOPDIR line from the spec file, but that would mean accessing grep (and possibly some other tool like sed, awk or cut) before we have properly checked for that, and we have not done anything like that so early up until now, so that was not attractive either.